documentation/phased-ranking.html

```diff
@@ -0,0 +1,127 @@
+after matching and before information is returned to the container.
+The number of hits to rerank can be configured as well in the rank-profile.
+</p>
+
+Example:
+
+<pre>
+search myapp {
+    &hellip;
+    rank-profile title-freshness inherits default {
+        first-phase {
+            expression: nativeRank(title) + freshness(timestamp)
+        }
+        second-phase {
+            expression {
+                xgboost("my_model.json") 
+            }
+            rerank-count: 50
+        }
+    }
+}
+</pre>
+In this example, the first phase uses the text matching feature <a href="reference/nativerank.html">nativeRank</a> scoped to the <em>title</em> field 
+plus one of the built-in document <a href="reference/rank-features.html">ranking features</a> named <em>freshness</em> over a <em>timestamp</em> field which stores the epoch time in second resolution.
+ For each content node the top 50 hits from
+the first phase function is re-ranked using a
+trained <a href="xgboost.html">xgboost</a> model. 
+</p>
+
+<h2 id="wand">Top-K Query Operator/Weak And</h2>
+<p>
+If the first-phase ranking function can be approximated as a simple linear function and the query mode is <em>any</em> the 
+<a href="https://dl.acm.org/citation.cfm?id=956944">Weak And/WAND</a> implementations in Vespa allows avoiding fully evaluating all the documents matching the query with the <em>first-phase</em> function. Instead 
+only the Top N hits using the internal operator ranking will be exposed to the <em>first-phase</em> ranking function.</p>
+
+<p>
+Both wand implementations accepts a targeted number of hits (top N) and each
+content node will expose all hits that were evaluated to the first-phase ranking function while skipped documents will not and the skipped set are the ones that matches the query but which cannot outperform any of
+the already collected hits on the internal scoring heap.
+Due to not fully evaluating all documents matching the query the total hit count becomes inaccurate when using weakAnd/Wand.</p> 
+
+<p>See using <a href="using-wand-with-vespa.html">Wand with Vespa</a> documentation</p>
+
+<h2 id="choosing-functions">Choosing phased ranking functions </h3>
+<p>
+A good ranking expression will for most applications consume too much CPU
+to be runnable on the entire result set within the latency budget/SLA,
+so in most cases the desired rank function should be used as the second phase function.
+The task then becomes to find a first phase function,
+which correlates sufficiently well with the second phase function,
+to ensure that relevance is not hurt too much by not evaluating the second phase function on all the hits.
+</p>
+
+
+<h2 id="evaluation">Evaluating relevancy impact of phased ranking</h2>
+<p>
+The impact of using a cheaper function in the first phase can be assessed by deploying
+the second order function as the first order function in a test rank profile,
```

Comment:
Consider using 'phase' instead of 'order' in this section as used elsewhere in the document.